ydb/docs/en/core/integrations/grafana.md

```diff
@@ -0,0 +1,132 @@
+
+```typescript
+    jsonData:
+      authKind: "Anonymous" | "ServiceAccountKey" | "AccessToken" | "UserPassword" | "MetaData";
+      endpoint: string;
+      dbLocation: string;
+      user?: string;
+    secureJsonData:
+      serviceAccAuthAccessKey?: string;
+      accessToken?: string;
+      password?: string;
+      certificate?: string;
+```
+
+## Building queries
+
+[YQL dialect](../yql/reference/index.md) is used to query YDB.
+Queries can contain macros which simplify syntax and allow for dynamic parts.
+The query editor allows you to get data in different representation: time series, table or logs.
+
+### Time series
+
+Time series visualization options are selectable after adding to your query one field with `Date`, `Datetime` or `Timestamp` type and at least one field with `number` type. You can select time series visualizations using the visualization options. Grafana interprets timestamp rows without explicit time zone as UTC. Any other column is treated as a value column.
+
+#### Multi-line time series
+
+To create multi-line time series, the query must return at least 3 fields in the following order:
+
+- field 1: time field
+- field 2: value to group by
+- field 3+: the metric values
+
+For example:
+
+```sql
+SELECT `timestamp`, `requestTime`, AVG(`responseStatus`) AS `avgRespStatus`
+FROM `/database/endpoint/my-logs`
+GROUP BY `requestTime`, `timestamp`
+ORDER BY `timestamp`
+```
+
+### Tables
+
+Table visualizations will always be available for any valid YDB query.
+
+### Visualizing logs with the Logs Panel
+
+To use the Logs panel your query must return a time and string values. You can select logs visualizations using the visualization options.
+
+By default only the first text field will be represented as log line, but this can be customized using query builder.
+
+### Macros
+
+To simplify syntax and to allow for dynamic parts, like date range filters, the query can contain macros.
+
+Here is an example of a query with a macro that will use Grafana's time filter:
+
+```sql
+SELECT `timeCol`
+FROM `/database/endpoint/my-logs`
+WHERE $__timeFilter(`timeCol`)
+```
+
+| Macro                                        | Description                                                                                                                      | Output example                                                                                  |
+| -------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------- |
+| _$\_\_timeFilter(columnName)_                | Replaced by a conditional that filters the data (using the provided column) based on the time range of the panel in microseconds | `foo >= CAST(1636717526371000 AS TIMESTAMP) AND foo <=  CAST(1668253526371000 AS TIMESTAMP)' )` |
```

Comment:
does it indeed support only columns, or expressions are allowed too?

Reply:
Only column name may be used here.

Reply:
Then a hint saying to use a subquery if you need an expression would be useful.

Reply:
sorry, I can't catch what do you want to see here :(